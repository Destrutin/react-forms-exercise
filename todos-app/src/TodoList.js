import React, { useState } from "react";
import Todo from "./Todo"
import NewTodoForm from "./NewTodoForm"; 

function TodoList() {
    const [todos, setTodos] = useState([]);

    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, newTodo]);
    }

    const remove = (id) => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    }
    console.log("Todos:", todos);
    return (
        <>
            <h1>Todo List</h1>
            <NewTodoForm addTodo={addTodo}/>
            <div>
                <ul>
                    {todos.map(({ id, formData }) => 
                        <Todo 
                            id={id} 
                            key={id} 
                            todo={ formData }
                            removeTodo={remove}
                            data-testid={`todo-${id}`}
                    />)}
                </ul>
            </div>
        </>
    )
}

export default TodoList;