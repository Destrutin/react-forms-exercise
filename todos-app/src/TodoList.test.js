import React from "react";
import { render, fireEvent} from "@testing-library/react";
import TodoList from "./TodoList";

test("renders without crashing", () => {
    render(<TodoList/>);
})

test("matches snapshot", () => {
    const {asFragment } = render(<TodoList/>);
    expect(asFragment()).toMatchSnapshot();
})

test("should add a new todo", () => {
    const { queryByText, getByLabelText, getByTestId } = render(<TodoList/>);
    const todo = getByLabelText("Todo");
    const btn = queryByText("Add Todo");
    
    fireEvent.change(todo, { target: { value: "test" }});
    fireEvent.click(btn);

    expect(getByTestId("todo")).toBeInTheDocument();
})

test("should remove a todo", () => {
    const { queryByText, getByLabelText, getByTestId } = render(<TodoList/>);
    const todo = getByLabelText("Todo");
    const btn = queryByText("Add Todo");
    
    fireEvent.change(todo, { target: { value: "test" }});
    fireEvent.click(btn);

    expect(getByTestId("todo")).toBeInTheDocument();

    const removeBtn = queryByText("X");
    fireEvent.click(removeBtn);
    expect(removeBtn).not.toBeInTheDocument();
})