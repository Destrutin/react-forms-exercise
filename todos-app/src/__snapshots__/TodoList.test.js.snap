// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Todo List
  </h1>
  <form>
    <label
      for="todo"
    >
      Todo
    </label>
    <input
      id="todo"
      name="todo"
      placeholder="New todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
  <div />
</DocumentFragment>
`;
